src/sumcheck/sumcheck_small_value.rs

```diff
@@ -0,0 +1,841 @@
+
+            // beta = (1,0,1)
+            local_accumulators.accumulate(0, 1, e0_1 * temp_acc[5]); // u = 1, y = 01
+            local_accumulators.accumulate(1, 2, e1_1 * temp_acc[5]); // v = 1, u = 0, y = 1
+            local_accumulators.accumulate(2, 5, e2 * temp_acc[5]); // v = 10, u = 1
+
+            // beta = (1,1,0)
+            local_accumulators.accumulate(0, 1, e0_2 * temp_acc[6]); // u = 1, y = 10
+            local_accumulators.accumulate(1, 3, e1_0 * temp_acc[6]); // v = 1, u = 1, y = 0
+            local_accumulators.accumulate(2, 6, e2 * temp_acc[6]); // v = 11, u = 0
+
+            // beta = (1,1,1)
+            local_accumulators.accumulate(0, 1, e0_3 * temp_acc[7]); // u = 1, y = 11
+            local_accumulators.accumulate(1, 3, e1_1 * temp_acc[7]); // v = 1, u = 1, y = 1
+            local_accumulators.accumulate(2, 7, e2 * temp_acc[7]); // v = 11, u = 1
+
+            local_accumulators
+        })
+        .reduce(|| Accumulators::<EF>::new_empty(), |a, b| a + b) 
+}
+
+// Given a point w = (w_1, ..., w_l), it returns the evaluations of eq(w, x) for all x in {0, 1}^l.
+pub fn eval_eq_in_hypercube<F: Field>(point: &Vec<F>) -> Vec<F> {
+    let n = point.len();
+    let mut out = F::zero_vec(1 << n); 
+    eval_eq_batch::<F, F, false>(
+        RowMajorMatrixView::new_col(&point),
+        &mut out,
+        &[F::ONE],
+    );
+    out 
+}
+
+// Given two multilinear points p and q, it evaluates eq(p, q).
+pub fn eval_eq_in_point<F: Field>(p: &[F], q: &[F]) -> F {
+    let mut acc = F::ONE;
+    for (&l, &r) in p.into_iter().zip(q) {
+        acc *= F::ONE + l * r.double() - l - r;
+    }
+    acc
+}
+
+// Given the points w and r, we compute the linear function 
+// l(x) = eq( w_1,...w_{i-1} ; r_1,...r_{i-1} ) * eq(w_i, x)
+// Section 5.1, page 17.
+pub fn compute_linear_function<F: Field>(w: &[F], r: &[F]) -> [F; 2] {
+    let round = w.len();
+    debug_assert!(r.len() == round - 1);
+
+    let mut const_eq: F = F::ONE;
+    if round != 1 {
+        const_eq = eval_eq_in_point(&w[..round - 1], r);
+    }
+    let w_i = w.last().unwrap();
+
+    // Evaluation of eq(w,X) in [eq(w,0),eq(w,1)]
+    [const_eq * (F::ONE - *w_i), const_eq * *w_i]
+}
+
+// Given the linear functions l and t, we compute S(0) and S(inf).
+// See Procedure 8, Page 35.
+fn get_evals_from_l_and_t<F: Field>(l: &[F; 2], t: &[F]) -> [F; 2] {
+    [
+        t[0] * l[0],                   // S(0)
+        (t[1] - t[0]) * (l[1] - l[0]), // S(inf) -> l(inf) = l(1) - l(0)
+    ]
+}
+
+// Algorithm 6. Page 19.
+// Compute three sumcheck rounds using the small value optimizaition and split-eq accumulators.
+// It returns the three challenges r_1, r_2, r_3.
+pub fn svo_three_rounds<Challenger, F: Field, EF: ExtensionField<F>>(
+    prover_state: &mut ProverState<F, EF, Challenger>,
+    poly: &EvaluationsList<F>,
+    w: &MultilinearPoint<EF>,
+    sum: &mut EF,
+) -> (EF, EF, EF)
```

Comment:
Here shouldn't we have `pow_bits` as an argument as well to do grinding for each round?

```suggestion
pub fn svo_three_rounds<Challenger, F: Field, EF: ExtensionField<F>>(
    prover_state: &mut ProverState<F, EF, Challenger>,
    poly: &EvaluationsList<F>,
    w: &MultilinearPoint<EF>,
    sum: &mut EF,
    pow_bits: usize,
) -> (EF, EF, EF)
```